lib/chat_api_web/controllers/note_controller.ex

```diff
@@ -0,0 +1,65 @@
+defmodule ChatApiWeb.NoteController do
+  use ChatApiWeb, :controller
+
+  alias ChatApi.Notes
+  alias ChatApi.Notes.Note
+
+  action_fallback ChatApiWeb.FallbackController
+
+  @spec index(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def index(conn, %{"note" => note_params}) do
+    %{"customer_id" => customer_id} = note_params
+
+    with %{account_id: account_id} <- conn.assigns.current_user do
+      notes = Notes.list_notes_for_customer(%{account_id: account_id, customer_id: customer_id})
+      render(conn, "index.json", notes: notes)
+    end
+  end
+
+  @spec create(Plug.Conn.t(), map()) :: Plug.Conn.t()
+  def create(conn, %{"note" => note_params}) do
+    %{"body" => body, "customer_id" => customer_id} = note_params
+
+    with %{account_id: account_id, id: author_id} <- conn.assigns.current_user do
+      with {:ok, %Note{} = note} <-
```

Comment:
no need to nest `with` statements :) we can just do:
```elixir
with %{account_id: account_id, id: author_id} <- conn.assigns.current_user,
  {:ok, %Note{} = note} <- Notes.create_note(%{...}) do
  # etc
end
```

(some docs on `with`: https://elixirschool.com/en/lessons/basics/control-structures/#with)

Reply:
Ah: with is a context (return context), not an extraction operator (that's just the `<-` part) 💡

Reply:
Huh, it took me a while but I think I get this..

So
```
with %{account_id: account_id, id: author_id} <- conn.assigns.current_user,
  {:ok, %Note{} = note} <- Notes.create_note(%{...}) do
```

if the {:ok, %Note{}} doesn't match (for ex, if create_note returns an {:error, Changeset}), it just falls through to outside the `with` (and in this case is handled by our FallbackController)?